src/Biogeochemistry.jl

```diff
@@ -0,0 +1,138 @@
+module Biogeochemistry
+
+using Oceananigans.Grids: Center, xnode, ynode, znode
+using Oceananigans.Advection: div_Uc, CenteredSecondOrder
+using Oceananigans.Utils: tupleit
+using Oceananigans.Fields: Field
+using Oceananigans.Architectures: device, architecture
+using Oceananigans.Utils: launch!
+using KernelAbstractions: @kernel
+
+import Oceananigans.Fields: location, CenterField
+import Oceananigans.Forcings: regularize_forcing, maybe_constant_field
+
+#####
+##### Generic fallbacks for biogeochemistry
+#####
+
+"""
+Update tendencies.
+
+Called at the end of calculate_tendencies!
+"""
+update_tendencies!(bgc, model) = nothing
+
+"""
+Update tracer tendencies.
+
+Called at the end of calculate_tendencies!
+"""
+update_biogeochemical_state!(bgc, model) = nothing
+
+@inline biogeochemical_drift_velocity(bgc, val_tracer_name) = nothing
+@inline biogeochemical_advection_scheme(bgc, val_tracer_name) = nothing
```

Comment:
why do we need this and not just a scheme for the tracer?

Reply:
I don't know if there was a reason for this @glwagner, but it would simplify making a bgc model for a user if we just used `model.advection`?

Reply:
I guess the trade-off here is between simplicity and flexibility. Do you envision the biological tracer having any particular advection scheme different than the tracer that needs to be stored in the bcg model?

Reply:
Hmm, ok I see what you mean. If stability is the reasoning I would argue in favor of removing the additional advection scheme also in `AdvectiveForcing`. 

There should be no reason for a velocity to be treated in a `special` way, as the reconstruction (especially if using upwinding) applies to the tracer, not to the velocity.

If there is instability due to dispersion and your tracer is advected in two different ways it might be difficult (and a headache) to understand which of the "different" velocities is causing it.

Reply:
It might actually be even better to just sum the velocities so that the whole upwinding is consistent if different velocities have different signs.

Reply:
I see what you mean. I think it would take quite a lot of restructuring of the tendency kernels to impliment advective forcing in this way? Based on this I would also guess it would be better to change how backgound velocities are added since this will also be upwinded incorrectly:
```julia
 - div_Uc(i, j, k, grid, advection, velocities, c)
 - div_Uc(i, j, k, grid, advection, background_fields.velocities, c)
```

I'm happy to try and do this but it might make sense as a different PR as it would be quite a lot of changes?

Reply:
yeah... I guess advecting with like this something might suffice
```
julia> struct Utot{U, Ub, Ubgc}
       u           :: U
       uBackground :: Ub
       uBiogeochem :: Ubgc
       end

julia> Base.getindex(u::Utot, i, j, k) = u.U[i, j, k] + u.ubackgound[i, j, k] + u.uBiogechem[i, j, k]
```
but you don't have to do it here.

Reply:
I ll open an issue for that

Reply:
I prefer the design we have

Reply:
I agree it'd be better to combine the velocities. The reason we haven't tried that yet is because I regard these as experimental features. So I'd like to let the design mature a bit before adding that complication. I suggest we keep the velocities separate for now.

We'll also have to discuss how to name these velocities.

Reply:
ok, then we should add a warning to not use upwinding. From what I understood false extrema and non-bounds-preserving algorithms are very detrimental in biogeochemistry models

Reply:
Can we leave it for now and develop a solution after this PR is merged?